actions/parse-push/src/main.ts

```diff
@@ -64,49 +70,59 @@ async function getSlug(commit: any): Promise<string> {
  *             Due to the nature of problem-specs, this will generally hold true.
  *
  * @returns The PR associated with the commit.
- * @param commit_sha
+ * @param commitSha
  */
-export async function getPullRequestHTMLURL(commit_sha: string): Promise<string> {
-    const octokit = gh.getOctokit(core.getInput('token'))
-    const pullCandidates = await octokit.repos.listPullRequestsAssociatedWithCommit({
-        owner: gh.context.repo.owner,
-        repo: gh.context.repo.repo,
-        commit_sha: commit_sha,
-        mediaType: {
-            previews: ['groot'], // This endpoint is part of a preview
-        },
-    })
+export async function getPullRequestHtmlUrl(
+    commitSha: string
+): Promise<string> {
+    const octokit = getOctokit(core.getInput('token'))
+    const pullCandidates = await octokit.repos.listPullRequestsAssociatedWithCommit(
+        {
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            commit_sha: commitSha,
+            mediaType: {
+                previews: ['groot'], // This endpoint is part of a preview
+            },
+        }
+    )
 
     // Filter out unmerged PRs and PRs with a non-default base branch
-    const pulls = pullCandidates.data.filter(x => {
-        const eventPayload = gh.context.payload as Webhooks.EventPayloads.WebhookPayloadPush // Otherwise TS will moan about default_branch possibly being undefined
-        return x.base.ref == `${eventPayload.repository.default_branch}` && x.state == 'closed' && x.merge_commit_sha
+    const pulls = pullCandidates.data.filter((x) => {
+        const eventPayload = context.payload
+
+        return (
+            eventPayload.repository &&
+            x.base.ref == `${eventPayload.repository.default_branch}` &&
+            x.state == 'closed' &&
+            x.merge_commit_sha
+        )
     })
 
     // Assumption: Commits only occur once on the default branch
     if (pulls.length != 1) {
         core.warning(
-            `Could not determine a PR associated with ${commit_sha}; more than one candidate: ${pulls}. Expected a unique association.`,
+            `Could not determine a PR associated with ${commitSha}; more than one candidate: ${pulls}. Expected a unique association.`
         )
         return ''
     }
 
     return pulls[0].html_url
 }
 
+// uuid4 regex from https://stackoverflow.com/questions/11384589/what-is-the-correct-regex-for-matching-values-generated-by-uuid-uuid4-hex/18516125#18516125
+const UUID_V4_PATTERN = /^\+\s*"uuid": "([a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12})"/
```

Comment:
Technically the uuid4 pattern is just `[a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12}`. The rest of the pattern is used to find added UUIDs in the diff.

Is it possible to interpolate regex strings?

Reply:
Sorta yes. RegExp can be created from a string using `new RegExp(string, flags)`.

Reply:
```suggestion
const ADDED_UUID_PATTERN = /^\+\s*"uuid": "([a-f0-9]{8}-?[a-f0-9]{4}-?4[a-f0-9]{3}-?[89ab][a-f0-9]{3}-?[a-f0-9]{12})"/
```

Perhaps just change the name to clarify what it is rather than constructing a Regex from two snippets then?

Reply:
Yeah, I'm good with that!